cli/src/commands/GenerateCommand.ts

```diff
@@ -195,8 +202,11 @@ export class GenerateCommand extends Command {
             owner: this.owner,
             repository: this.repository,
             specPath: this.specPath,
-          };
+          } as GithubOptions;
         }
+
+      default:
+        throw new UsageError("--source argument is invalid");
```

Comment:
I had to throw an error for the return type of the function to be correct.

It was `FromOptions | undefined` before. Also, the default case is important.